docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+import os
+
+# Get the engine instance that is currently running.
+current_engine = sgtk.platform.current_engine()
+
+# Grab the pre-created Sgtk instance from the current engine.
+tk = current_engine.sgtk
+
+# Get a context object from a Task, this Task must belong to a Shot for the future steps to work. 
+context = tk.context_from_entity("Task", 13155)
+
+# Create the required folders based upon the task.
+tk.create_filesystem_structure("Task", context.task["id"])
+
+# Get a template instance by providing a name of a valid template in your config's templates.yml.
+template = tk.templates["maya_shot_publish"]
+
+# Use the context to resolve as many of the template fields as possible.
+fields = context.as_template_fields(template)
+
+# Manually resolve the remaining fields that can't be figured out automatically from context.
+fields["name"] = "myscene"
+fields["version"] = 1
+
+# Use the fields to resolve the template path into an absolute path.
+publish_path = template.apply_fields(fields)
+
+# Make sure we create any missing folders.
+current_engine.ensure_folder_exists(os.path.dirname(publish_path))
+
+# Create a empty file on disk. (optional - should be replaced by actual file save or copy logic)
+sgtk.util.filesystem.touch_file(publish_path)
+```
+
+The next step is to dynamically work out the next version number rather than hard coding it.
+
+## Part 6 - Finding existing files and getting the latest version number
+
+There two methods you could use here. 
+
+1. Since in this particular example you are resolving a publish file, you could use the Shotgun API to query for the
+next available version number on `PublishedFile` entities.
+2. You can scan the files on disk and work out what versions already exist, and extract the next version number. 
+This is helpful if the files you're working with aren't tracked in Shotgun (such as work files).
+
+Whilst the first option would probably be most suitable for the example in this guide, both approaches have their uses so we'll cover them both.
+
+### Querying Shotgun for the next version number.
+
+Using the SG API and the [`summarize()` method](https://developer.shotgunsoftware.com/python-api/reference.html#shotgun_api3.shotgun.Shotgun.summarize) we can get the highest version number amongst 
+the PublishedFiles with the same name on the task, and then add 1.
+
+```python
+r = sg.summarize(entity_type="PublishedFile",
+                 filters = [["task", "is", {"type":"Task", "id": context.task["id"]}],
+                            ["name","is", fields["name"] + ".ma"]],
+                 summary_fields=[{"field":"version_number", "type":"maximum"}])
+
+# Extract the version number and add 1 to it.
+# In scenarios where there are no files already this summary will return 0.
+fields["version"] = r["summaries"]["version_number"] + 1
+```
+
+{% include info title="Note" content="To save writing out the name twice, you can take it from the fields dictionary you defined in the previous step." %}
```

Comment:
you need to provide the name of the file in the filters for the summarize, I opted to take the name from the `fields["name"]` instead of hard coding it here as well.

Reply:
lets remove the note